### ISSUE TITLE
ng test losing typings after retrigger

### ISSUE THREAD

**Issue by bialad** [angular/angular-cli#5332]
### OS?
Linux (Ubuntu 16.04)

### Versions.
@angular/cli: 1.0.0-rc.1
node: 6.10.0
os: linux x64
@angular/common: 2.4.9
@angular/compiler: 2.4.9
@angular/core: 2.4.9
@angular/forms: 2.4.9
@angular/http: 2.4.9
@angular/platform-browser: 2.4.9
@angular/platform-browser-dynamic: 2.4.9
@angular/router: 3.4.9
@angular/cli: 1.0.0-rc.1
@angular/compiler-cli: 2.4.9

### Repro steps.
- ng new ngTest
- cd ngTest
- yarn
- Create folder typings in src
- Create file common.d.ts in src/typings with content:
```
declare module Common {
  export interface IAction {
    type: string;
  }
}
```
- In app.component.ts add a constructor
```
constructor() {
  const action: Common.IAction = { type: 'test' };
}
```
- Running ng test once works
- Saving app.component.ts (retriggering ng test) fails to find type definition, but test succeeds

### The log given by the failure.
```
ERROR in /home/tobias/projects/external/ngTest/src/app/app.component.ts (12,19): Cannot find namespace 'Common'.)
Chrome 56.0.2924 (Linux 0.0.0): Executed 3 of 3 SUCCESS (0.222 secs / 0.215 secs)
```

**Comment by bialad** [comment]
After fiddling some more with this I notice that the typing error actually stops the rebuild, and runs the test again with the previous build.

This make it look like the test succeeds even though the typing fails, but it's just showing a result from a previous build.

To be able to run tests that actually tests my current code, I need to use `ng test --single-run true`.

**Comment by bialad** [comment]
I'm sorry to say that it still doesn't work with the application I created for this issue. I updated to rc4 as shown here:

```
    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
@angular/cli: 1.0.0-rc.4
node: 6.10.0
os: linux x64
@angular/common: 2.4.10
@angular/compiler: 2.4.10
@angular/core: 2.4.10
@angular/forms: 2.4.10
@angular/http: 2.4.10
@angular/platform-browser: 2.4.10
@angular/platform-browser-dynamic: 2.4.10
@angular/router: 3.4.10
@angular/cli: 1.0.0-rc.4
@angular/compiler-cli: 2.4.10
```

The change I see is that the test doesn't succeed, which is certainly an improvement. :)

I'm I doing something wrong by adding a d.ts file like this. `ng serve` and `ng build` works fine with this setup, just like before.

**Comment by filipesilva** [comment]
@bialad

**Comment by filipesilva** [comment]
Even though that change was merged a couple of days ago

**Comment by bialad** [comment]
@filipesilva The results are different now, but still not correct.

Even though it's a standard cli application, I created a [`repo`](https://github.com/bialad/ng-test-typings) to base the discussion on. I recreated the application from scratch with the CLI 1.0.0 and Anuglar 4.0.0 version, to be sure I'm not encountering any bugs because of mismatching version when updating.

- Now when I run `ng test` the first time, everything work. As it did before.

- Then changing the code in [app.component.ts](https://github.com/bialad/ng-test-typings/blob/master/src/app/app.component.ts) behaves as expected. For example changing `'test'` to `1` fails. Great so far. :)

- If I keep `'test'` in AppComponent, and instead change the type property in [common.d.ts](https://github.com/bialad/ng-test-typings/blob/master/src/typings/common.d.ts) from `string` to `number`, it successes. Not good.

- If I keep the type change and stop and restart `ng test` it still succeeds, which it shouldn't...

- Then changing `'test'` i AppComponent to `1` succeeds (as it now should), BUT when changing `1` back to `'test'` it falis (as it should).

This is very confusing to me, especially when the behavior isn't consistent. My guess is some sort of type caching not updating correctly...

Hope you could follow my steps here, and am able to reproduce this issue. Otherwise it must be something in my platform. The fact the I can now run my tests without `single-run` as long as I don't change my typings is i big step forward and a time saver.

**Comment by boldwade** [comment]
I'm experiencing a similar issue as well. I'm on cli, 1.0.0-rc.2 but our ng version is at 2.4.9. Karma is 1.5, typescript is 2.1.6. Running a test the first time works as expected, make a simple change to kick off another test run and the build will randomly fail with some random typescript error, such as 'HTMLInputElement' not found, for what appears to be a problem with the typings get lost. This doesn't appear to be an issue that occurs every time, seems to be a bit random. I've been noticing this off and on for about a week. Not sure the root cause, but the above sounds like a match.

**Comment by filipesilva** [comment]
@bialad ah, that exist exists but it's a different one! https://github.com/angular/angular-cli/issues/5634 seems to be related to interfaces and probably all `.d.ts` files instead. We're tracking it there instead.

@boldwade unfortunately to fix this you really need to update to `1.0.0`, that's when the fix landed.
